include/ginkgo/core/base/executor.hpp

```diff
@@ -104,12 +104,37 @@ constexpr allocation_mode default_hip_alloc_mode =
 inline namespace cl {
 namespace sycl {
 
+
 class queue;
 
+
 }  // namespace sycl
 }  // namespace cl
 
 
+/**
+ * The enum class for the dpcpp queue property.
+ */
+enum class dpcpp_queue_property {
+    /**
+     * queue executed in order
+     */
+    in_order = 1,
+
+    /**
+     * queue enables the profiling
+     */
+    enable_profiling = 2
+};
+
+GKO_ATTRIBUTES GKO_INLINE dpcpp_queue_property operator|(dpcpp_queue_property a,
+                                                         dpcpp_queue_property b)
+{
+    return static_cast<dpcpp_queue_property>(static_cast<int>(a) |
+                                             static_cast<int>(b));
+}
```

Comment:
I add one enum item for 1 | 2. I do not like this way because it is hard to extend for more options by providing all possible combination. Do you have any idea for it?
I only think store it in integer and provide the operation/assignment between enum and integer such that user can use enum to select the property.

Reply:
IMO using enums for flags is well-established, so I'd be fine with keeping just the two entries.

Reply:
I had to do a lot of reading to understand what the standard says, as [cppreference](https://en.cppreference.com/w/cpp/language/enum) was IMO not clear about this.
According to C++14 ([here is the open draft](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2013/n3797.pdf)) [dcl.enum] p8:
> For an enumeration whose underlying type is fixed, the values of the enumeration are the values of the
underlying type.

It is legal for an enum type to have a value that is not part of the enumeration (which I didn't know).
That means, your original implementation was fine and I was wrong, sorry for that.

However, as a small nit, I would like to have a comment for the enum definition itself that specifies that the enumerators are allowed to be used in combination by performing a bitwise or (`|`).

Reply:
thanks!